scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/synchronization/SynchronizationInternal.java

```diff
@@ -0,0 +1,144 @@
+
+    Set<Map.Entry<String, Serializable>> entrySet(String originator, TaskId taskid, String channel)
+            throws InvalidChannelException;
+
+    Serializable getOrDefault(String originator, TaskId taskid, String channel, String key, Serializable defaultValue)
+            throws InvalidChannelException;
+
+    void forEach(String originator, TaskId taskid, String channel, String action)
+            throws InvalidChannelException, CompilationException;
+
+    void replaceAll(String originator, TaskId taskid, String channel, String function)
+            throws InvalidChannelException, CompilationException, IOException;
+
+    Serializable putIfAbsent(String originator, TaskId taskid, String channel, String key, Serializable value)
+            throws InvalidChannelException, IOException;
+
+    boolean remove(String originator, TaskId taskid, String channel, String key, Serializable value)
+            throws InvalidChannelException, IOException;
+
+    boolean replace(String originator, TaskId taskid, String channel, String key, Serializable oldValue,
+            Serializable newValue) throws InvalidChannelException, IOException;
+
+    Serializable replace(String originator, TaskId taskid, String channel, String key, Serializable value)
+            throws InvalidChannelException, IOException;
+
+    Serializable computeIfAbsent(String originator, TaskId taskid, String channel, String key, String mappingFunction)
+            throws InvalidChannelException, CompilationException, IOException;
+
+    Serializable computeIfPresent(String originator, TaskId taskid, String channel, String key,
+            String remappingFunction) throws InvalidChannelException, CompilationException, IOException;
+
+    Serializable compute(String originator, TaskId taskid, String channel, String key, String remappingFunction)
+            throws InvalidChannelException, CompilationException, IOException;
+
+    Serializable merge(String originator, TaskId taskid, String channel, String key, Serializable value,
+            String remappingFunction) throws InvalidChannelException, CompilationException, IOException;
+
+    // Additional operations
+
+    PredicateActionResult conditionalCompute(String originator, TaskId taskid, String channel, String key,
+            String predicate, String thenRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException;
+
+    PredicateActionResult conditionalCompute(String originator, TaskId taskid, String channel, String key,
+            String predicate, String thenRemappingFunction, String elseRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException;
+
+    void conditionalReplaceAll(String originator, TaskId taskid, String channel, String predicate,
+            String thenRemappingFunction) throws InvalidChannelException, CompilationException, IOException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)